dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala

```diff
@@ -53,13 +53,75 @@ class CMul[@specialized(Float, Double) T: ClassTag](
       } else {
         weight.view(Array(1) ++ weight.size())
       }
-
-      expand.expandAs(output)
-      output.cmul(expand)
+      val pivotDim = getOnlyDimGtOne(expand.size())
+      if (pivotDim > 0) {
+        mulOneDimWeight(pivotDim, expand, output)
+      } else {
+        expand.expandAs(output)
+        output.cmul(expand)
+      }
     }
     output
   }
 
+  /**
+   * if there is only one dim of size > 1, return this dim(count from 1)
+   * else return -1
+   * e.g. (1, 2, 1, 1) returns 1, (1, 2, 3, 1) returns -1, and (1, 1, 1, 1) returns -1
+   * @param size size of tensor
+   * @return (the only dim whose value > 1) else (-1)
+   */
+  private def getOnlyDimGtOne(size: Array[Int]): Int = {
```

Comment:
is it same with CAdd? if so, can we put them into a utils class instead of two copies of code?